src/data/markdown/translated-guides/en/02 Using k6/08 Tags and Groups.md

```diff
@@ -96,6 +95,160 @@ export default function () {
 
 </CodeGroup>
 
+## Test-wide tags
+
+Besides attaching tags to requests, checks, and custom metrics, you can set test-wide tags across all metrics.
+There are two ways you can set these tags:
+
+- In the CLI, using one or more `--tag NAME=VALUE` flags
+
+- In the script itself:
+
+  <CodeGroup labels={["test-wide-tags.js"]} lineNumbers={[true]}>
+
+  ```javascript
+  export const options = {
+    tags: {
+      name: 'value',
+    },
+  };
+  ```
+
+  </CodeGroup>
+
+## Code-defined tags
+
+In the case, a user-defined tag with advanced logic for handling which tag to set is required then it's possible doing it by defining the tag from the code.
+
+It's possible to use the [k6/execution.vu.tags](/javascript-api/k6-execution/#vu) API for assigning an object property with the name and the value of the required tag. Like the following example for tracking the container group from the nested groups, it would be helpful for aggregating sub-metrics for nested groups.
```

Comment:
```suggestion
To support advanced tagging logics and workflows, it is also possible to directly manipulate them from scripts' code.

[k6/execution.vu.tags](/javascript-api/k6-execution/#vu) object's properties can indeed be directly assigned new key/value pairs to define new tags dynamically. This can prove useful, as demonstrated in the following example, to track a container's group from nested groups, and aggregating nested group's sub-metrics.
```

Here's a proposal I'm not too happy with, as I was not 100% sure I understand what you meant in detail. Please don't hesitate to adjust accordingly. On a sidenote: I encountered the issue recently, and I don't believe we even document sub-metrics publicly?